ONBOARDING: Dialog with character pressed

priettt opened the pull request:
When clicking a character, a dialog shows up with the character's information

Signed-off by: Francisco Prieto francisco.prieto@globant.com

Commit:
ONBOARDING: Dialog with character pressed

When clicking a character, a dialog shows up with the character's information

Signed-off by: Francisco Prieto francisco.prieto@globant.com

SartiMau commented on app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt:
```diff
@@ -1,16 +1,45 @@
+                val subscription = getCharacterDetailServiceUseCase.invoke(value.id).
+                        subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+                        .subscribe({ character ->
+                    var detailsLink = ""
```
use constants please

eis6aer replied:
I would put the dot on the new line 
```suggestion
                    getCharacterDetailServiceUseCase.invoke(value.id)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
```

priettt replied:
done!

SartiMau commented:
Please add a video showing how the app is working with the new functionality

eis6aer commented:
I would put the observable operators on a new line each 
```getCharacterDetailServiceUseCase.invoke(value.id)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())

Commit:
Code style fixes

Added constants and multi-line declarations

Signed-off by: francisco.prieto@globant.com

SartiMau commented on app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/adapter/viewholder/viewholder.kt:
```diff
@@ -13,7 +13,7 @@ class CharactersAdapterViewHolder(view: View, val listener: CharacterListener) :
     fun bind(item: Character) = with(itemView) {
         tv_item.text = item.name
         val string = item.thumbnail.path + "." + item.thumbnail.extension
```
no use constants and use strings templates pls

SartiMau commented on app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/extension/ImageExtensions.kt:
```diff
@@ -3,6 +3,11 @@ package com.puzzlebench.cleanMarvelKotlin.presentation.extension
 }
+
+fun ImageView.getImageByUrl(url: String?) {
+    if (url != null) Picasso.get().load(url).into(this)
```
use braces and put each function in a separated line pls

Commit:
Coding style fixes

Commit:
Downgraded Picasso and fixed unit tests

fanky10 reviewed (approved):
LG2M
